.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,75 @@
+name: Docker Build
+
+on:
+  schedule:
+    # Schedule runs run off HEAD commit on master.
+    # Run once a day to keep yunojuno/heroku up to
+    # date with Heroku's base image.
+    - cron:  '* 23 * * *'
+
+  push:
+    branches:
+      - master
+
+  pull_request:
+    types: [opened, synchronize, reopened]
```

Comment:
Is this correct - if we're creating a new build should it not run on `synchronize` only (assuming that's a merge)?

Reply:
Ah. `synchronize` is new commit on PR'd branch.

So this says:

* run on pushes to master
* run on PRs (on open event, on new commit event, on reopen event)
* run on the master branch at the 23rd hour every day (I'm not sure in which timezone actually but I assume UTC)

Reply:
Also note that creating a new build, and pushing said build are separate activities.

On PRs we want to run the build only - to ensure it completes. But when the `github.ref` is `master` (meaning it's a new commit on master branch or a scheduled run), we want to run the build _and_ push the image so it can be later drawn down.

Just running the build itself creates the image locally on the GA dyno, which is killed at the end of the CI run.